src/test/scala/chiselTests/Assert.scala

```diff
@@ -61,6 +61,17 @@ class BadUnescapedPercentAssertTester extends BasicTester {
   stop()
 }
 
+class PrintableFormattedAssertTester extends BasicTester {
+  val foobar = Wire(UInt(32.W))
+  foobar := 123.U
+  assert(foobar === 123.U, p"Error! Wire foobar =/= ${Hexadecimal(foobar)} This is 100%% wrong.\n")
+  stop()
+}
+
+class PrintableBadUnescapedPercentAssertTester extends BasicTester {
+  assert(1.U === 1.U, p"I'm 110% sure this is an invalid message")
```

Comment:
Actually this is valid, this is part of the reason we need to add the new String interpolator (ie. your `cf` PR), because it's perfectly valid to have `%` in a `p"..."` (eg. https://scastie.scala-lang.org/HdvWBGEERamX0v7yFc9CSw)

Reply:
Oh ok - I did not pay attention to this since I just merely copy-pasted what we had before and replaced the message with p".." - wonder how is the test passing then -  I think it is expecting a failure of some sorts